c2rust-transpile/src/translator/variadic.rs

```diff
@@ -9,130 +9,110 @@ pub enum VaPart {
-        // struct-based va_list (e.g. x86_64)
-        fn match_vastart_struct(ast_context: &TypedAstContext, expr: CExprId) -> Option<CDeclId> {
-            match_or! { [ast_context[expr].kind]
-            CExprKind::ImplicitCast(_, e, _, _, _) => e }
-            match_or! { [ast_context[e].kind]
-            CExprKind::DeclRef(_, va_id, _) => va_id }
+        let ast = &self.ast_context;
+
+        // `struct`-based `va_list` (e.g. x86_64).
+        let match_vastart_struct = || {
+            let e = match_or! { [ast[expr].kind]
+            CExprKind::ImplicitCast(_, e, _, _, _) => e }?;
+            let va_id = match_or! { [ast[e].kind]
+            CExprKind::DeclRef(_, va_id, _) => va_id }?;
             Some(va_id)
-        }
+        };
 
-        // struct-based va_list (e.g. x86_64) where va_list is accessed as a struct member
-        // supporting this pattern is necessary to transpile apache httpd
-        fn match_vastart_struct_member(
-            ast_context: &TypedAstContext,
-            expr: CExprId,
-        ) -> Option<CDeclId> {
-            match_or! { [ast_context[expr].kind]
-            CExprKind::ImplicitCast(_, me, _, _, _) => me }
-            match_or! { [ast_context[me].kind]
-            CExprKind::Member(_, e, _, _, _) => e }
-            match_or! { [ast_context[e].kind]
-            CExprKind::DeclRef(_, va_id, _) => va_id }
+        // `struct`-based `va_list` (e.g. x86_64) where `va_list` is accessed as a `struct` member.
+        //
+        // Supporting this pattern is necessary to transpile apache httpd.
+        let match_vastart_struct_member = || {
+            let me = match_or! { [ast[expr].kind]
+            CExprKind::ImplicitCast(_, me, _, _, _) => me }?;
+            let e = match_or! { [ast[me].kind]
+            CExprKind::Member(_, e, _, _, _) => e }?;
+            let va_id = match_or! { [ast[e].kind]
+            CExprKind::DeclRef(_, va_id, _) => va_id }?;
             Some(va_id)
-        }
+        };
 
-        // struct-based va_list (e.g. x86_64) where va_list is accessed as a member of a struct pointer
-        // supporting this pattern is necessary to transpile [graphviz](https://gitlab.com/graphviz/graphviz/-/blob/5.0.0/lib/sfio/sftable.c#L321)
-        fn match_vastart_struct_pointer_member(
-            ast_context: &TypedAstContext,
-            expr: CExprId,
-        ) -> Option<CDeclId> {
-            match_or! { [ast_context[expr].kind]
-            CExprKind::ImplicitCast(_, me, _, _, _) => me }
-            match_or! { [ast_context[me].kind]
-            CExprKind::Member(_, ie, _, _, _) => ie }
-            match_or! { [ast_context[ie].kind]
-            CExprKind::ImplicitCast(_, e, _, _, _) => e }
-            match_or! { [ast_context[e].kind]
-            CExprKind::DeclRef(_, va_id, _) => va_id }
+        // `struct`-based `va_list` (e.g. x86_64) where `va_list` is accessed as a member of a `struct *` pointer.
```

Comment:
"`struct *`" or "`struct` pointer" are fine ways to describe this pattern, but "`struct *` pointer" makes it sound like a double indirection could be present, which is not the case.

Reply:
True, I'll just use `struct *` then.

Reply:
Fixed in https://github.com/immunant/c2rust/pull/638/commits/8fae26f66fa8e21116cb3f8c8e646a50846152de.

Reply:
What we had before was perfectly fine. For instance
>  // `struct`-based `va_list` (e.g. x86_64) where `va_list` is accessed as a member of a `struct *`.

is not an improvement over the previous comment

>  // struct-based va_list (e.g. x86_64) where va_list is accessed as a member of a struct pointer

I'd prefer it if you can just revert to the old comments.

Reply:
I thought the `` ` ``s increased readability; they definitely do to me.  And there's no real risk of churn here, as it's just a comment.